src/rules/valid-mocked-module-path.ts

```diff
@@ -0,0 +1,99 @@
+import { statSync } from 'fs';
+import path from 'path';
+import { AST_NODE_TYPES, type TSESTree } from '@typescript-eslint/utils';
+import {
+  createRule,
+  findModuleName,
+  getAccessorValue,
+  isSupportedAccessor,
+  isTypeOfJestFnCall,
+} from './utils';
+
+export default createRule({
+  name: __filename,
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Disallow mocking of non-existing module path',
+    },
+    messages: {
+      invalidMockModulePath: 'Mocked module path {{moduleName}} does not exist',
+    },
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      CallExpression(node: TSESTree.CallExpression): void {
+        const { callee } = node;
+
+        if (callee.type !== AST_NODE_TYPES.MemberExpression) {
+          return;
+        }
+
+        const { property } = callee;
+
+        if (
+          !node.arguments.length ||
+          !isTypeOfJestFnCall(node, context, ['jest']) ||
+          !(
+            isSupportedAccessor(property) &&
+            ['mock', 'doMock'].includes(getAccessorValue(property))
+          )
+        ) {
+          return;
+        }
+
+        const [nameNode] = node.arguments;
+        const moduleName = findModuleName(nameNode);
```

Comment:
nit: likewise, this doesn't need its own variable

```suggestion
        const moduleName = findModuleName(node.arguments[0]);
```